### DISCUSSION TITLE
React-query "No QueryClient set, useQueryClientProvider to set one" storybook error

### DISCUSSION THREAD

**Discussion by ivanblank992** [storybookjs/storybook#17976] [Help]
Hi all,

In our app, we are using react-query to fetch and cache API data. But when it comes to creating a storybook component that displays that data I get a consistent error:

<img width="1510" alt="Screenshot 2022-04-15 at 11 21 34" src="https://user-images.githubusercontent.com/83581820/163553181-b769e095-4c9c-4c00-8a7b-eb0bbdd7abb4.png">

No matter what I do, there is no way to get rid of this error and make things work. I have tried a few approaches even by c/p these solutions:

1. https://github.com/storybookjs/storybook/issues/12489#issuecomment-702958192,
2. https://storybook.js.org/docs/react/writing-stories/build-pages-with-storybook#mocking-graphql-queries-with-msw-addon
3. https://msw-sb.vercel.app/?path=/story/demos-react-query--default-behavior

But no luck.

`import { QueryClient, QueryClientProvider } from 'react-query';

const queryClient = new QueryClient();

const withStoreProvider = Story => {
  return (
    <StoreProvider>
      <Story />
    </StoreProvider>
  
  );
};

const withReactQuery = Story => {
  return (
    <QueryClientProvider client={queryClient}>
      <Story />
    </QueryClientProvider>
  );
};

export const decorators = [withStoreProvider, withReactQuery];`

I have tried to wrap just a component that uses react-query and even  like this: 
` <StoreProvider>
      <QueryClientProvider client={queryClient}>
        <Story />
      </QueryClientProvider>
    </StoreProvider>`

Still getting the same error. If you need more details let me know. Does anyone have this problem?
Thanks

**Comment by jasmo2** [comment]
Did you solve it?

**Reply by livevsonline** [reply]
bump
